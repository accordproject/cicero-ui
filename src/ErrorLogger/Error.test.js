import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ErrorComponent from './Error';

const emptyFunction = jest.fn(() => {});

const mockedError = {
  clauseId: 'mockedId',
  parseError: {
    fileLocation: {
      end: {
        endColumn: 2,
        line: 1
      },
      start: {
        column: 1
      }
    },
    component: 'mockedComponent',
    fileName: 'mockedFileName',
    name: 'mockedName',
    shortMessage:
      'mockedShortMessage',
    message:
      'mockedLoginMessage'
  }
};

const mockedProps = { error: mockedError, errorNav: emptyFunction, errorProps: {} };

describe('<ErrorComponent />', () => {
  it('should render correct component', () => {
    const component = shallow(<ErrorComponent {...mockedProps} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('should show full message after clicking on arrow', () => {
    const component = shallow(<ErrorComponent {...mockedProps} />);
    component.find('ArrowDiv').simulate('click');
    expect(component.find('ErrorFullMessage').exists()).toBeTruthy();
  });
  it('should show full message after clicking on error type', () => {
    const component = shallow(<ErrorComponent {...mockedProps} />);
    component.find('ErrorType').simulate('click');
    expect(component.find('ErrorFullMessage').exists()).toBeTruthy();
  });
  it('should show full message after clicking on short message', () => {
    const component = shallow(<ErrorComponent {...mockedProps} />);
    component.find('ErrorShortMessage').simulate('click');
    expect(component.find('ErrorFullMessage').exists()).toBeTruthy();
  });
  it('should call errorNav function on click', () => {
    const component = shallow(<ErrorComponent {...mockedProps} />);
    component.find('ErrorFile').simulate('click');
    expect(emptyFunction).toHaveBeenCalledTimes(1);
  });
});
