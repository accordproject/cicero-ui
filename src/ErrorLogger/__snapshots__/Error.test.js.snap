// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorComponent /> should render correct component 1`] = `
<ErrorComponent>
  <ArrowDiv
    expanded={false}
    onClick={[Function]}
  />
  <ErrorFile
    onClick={[Function]}
  >
    Grammar
  </ErrorFile>
  <ErrorType
    onClick={[Function]}
  >
    mockedName
    :
  </ErrorType>
  <ErrorShortMessage
    onClick={[Function]}
  >
    mockedShortMessage
  </ErrorShortMessage>
</ErrorComponent>
`;
